src/elm/interval.coffee

```diff
@@ -251,7 +245,7 @@ module.exports.Collapse = class Collapse extends Expression
             a = b
         else if typeof b.low?.sameOrBefore == 'function'
           if a.high? and b.low.sameOrBefore doAddition(a.high, perWidth)
-            a.high = b.high if b.high.after a.high
+            a.high = b.high if !b.high? or b.high.after a.high
           else if a.high? and !b.high?
             a.high = b.high
```

Comment:
Just catching this potential issue now...  This seems wrong to me.  We're saying if a.high exists and b.high is null, then _even if b.low is after a.high_ set a.high to null.  For example:
```
A: Interval[2012-01-01T00:00:00.0, 2012-01-10T00:00:00.0]
B: Interval[2012-01-20T00:00:00.0, null]
Result: A collapse B
```
Based on my reading of the code, it seems like the result will be `Interval[2012-01-01T00:00:00.0, null]`.  Am I reading the code wrong?  Do we have a test like this to confirm correct behavior?